moon-store-mac-os/moon-store-mac-os/Domain/Repositories/ProductRepository.swift

```diff
@@ -0,0 +1,29 @@
+//
+//  ProductRepository.swift
+//  moon-store-mac-os
+//
+//  Created by Jose Luna on 12/23/24.
+//
+
+import Foundation
+
+final class ProductRepository: BaseNetworkService {
+    private let decoder: JSONDecoder
+    
+    init(decoder: JSONDecoder = .init()) {
+        self.decoder = decoder
+    }
+    
+    func getProducts() async throws -> [ProductModel] {
+        do {
+            let result = try await getData(for: MSEndpoint.products.path)
+            let productResponse = try decoder.decode(ProductResponse.self, from: result)
+            
+            if productResponse.data.isEmpty { throw MSError.noData }
+            
+            return productResponse.data
+        } catch let error as MSError {
+            throw error
+        }
```

Comment:
Bien si da un error que no es casteado correctamente? como problema de internet.
no hace un throw, a demas la funcion es async throw no de un error especifico, el casteo debe realizarse hasta que se necesite

```suggestion
        } catch {
            throw error
        }
```

Reply:
Los errores son casted to MSError en base

Reply:
Vas a estar casteando cada vez entonces?

Reply:
Error por solicitud. Mas controlado. Los errores externos a la app no.

Reply:
Esto genera los mismos problemas que te comente en un PR antiguo, no hay que asumir, siempre se debe agregar un caso por defecto que es lo que Brandon te menciona, podes mantener el catch del cast pero debajo de el tener un catch general para cualquier otro error.

Reply:
No tiene sentido estar casteando a cada rato esto, mejor siempre retornar solo el error

Reply:
En ello!